checkexperiments.py

```diff
@@ -0,0 +1,266 @@
+    out.write(initiating_run + '\nAward\tAccession\tcurrent status -> new status\tsubmitted date\n')
+    out.flush()
+    err.write(initiating_run + '\nAward\tAccession\terror message\n')
+    err.flush()
+    if bot_token:
+        sc = SlackClient(bot_token)
+        sc.api_call(
+            "chat.postMessage",
+            channel="#bot-reporting",
+            text=initiating_run,
+            as_user=True
+        )
+
+    # read_depth definitions:
+    min_depth = {}
+    min_depth['ChIP-seq'] = 20000000
+    min_depth['RAMPAGE'] = 10000000
+    min_depth['shRNA knockdown followed by RNA-seq'] = 10000000
+    min_depth['siRNA knockdown followed by RNA-seq'] = 10000000
+    min_depth['single cell isolation followed by RNA-seq'] = 10000000
+    min_depth['CRISPR genome editing followed by RNA-seq'] = 10000000
+    min_depth['modENCODE-chip'] = 500000
+
+    graph = []
+    # checkexperiments using a file with a list of experiment accessions to be checked
+    if accessions_list:
+        r = None
+        ACCESSIONS = []
+        if os.path.isfile(accessions_list):
+            ACCESSIONS = [line.rstrip('\n') for line in open(accessions_list)]
+        for acc in ACCESSIONS:
+            r = session.get(
+                urljoin(url, '/search/?field=@id&frame=object&limit=all&type=Experiment&accession=' + acc))
+            try:
+                r.raise_for_status()
+            except requests.HTTPError:
+                return
+            else:
+                local = copy.deepcopy(r.json()['@graph'])
+                graph.extend(local)
+    # checkexperiments using a query
+    else:
+        r = session.get(
+            urljoin(
+                url,
+                '/search/?type=Experiment' \
+                '&format=json&frame=object&limit=all&' + search_query))
+        try:
+            r.raise_for_status()
+        except requests.HTTPError:
+            return
+        else:
+            graph = r.json()['@graph']
+    print('number of experiments: ' + str(len(graph)))
+    for ex in graph:
+        experiment_accession = ex.get('accession')
+        award_request = session.get(urljoin(
+            url,
+            ex.get('award') + '?frame=object&format=json'))
+        award_obj = award_request.json()
+        replicates = set()
+        replicates_reads = {}
+        dates = []
+        files = []
+        try:
+            if ex.get('replicates'):
+                for replicate in ex.get('replicates'):
+                    replicate_request = session.get(urljoin(
+                        url,
+                        replicate + '?frame=object&format=json'))
+                    replicate_obj = replicate_request.json()
+                    if replicate_obj.get('status') not in ['deleted']:
+                        replicates.add(replicate_obj['@id'])
+                        replicates_reads[replicate_obj['@id']] = 0
+                if  ex.get('files'):
```

Comment:
extra space in the condition, after the if.

Reply:
done